script.js

```diff
@@ -1,12 +1,421 @@
 //You can edit ALL of the code here
-function setup() {
-  const allEpisodes = getAllEpisodes();
-  makePageForEpisodes(allEpisodes);
+
+function getData(URL) {
+  fetch(URL)
+    .then((response) => response.json())
+    .then((json) => {
+      makeShowPage(json);
+      selectShow(json);
+      search();
+      result();
+      home();
+      updateResult();
+      selectEpisode(json);
+    });
+}
+
+function setup(URL) {
+  fetch(URL)
+    .then((response) => response.json())
+    .then((json) => {
+      makeEpisodePage(json);
+      selectEpisode(json);
+    });
 }
 
-function makePageForEpisodes(episodeList) {
+function makeEpisodePage(list) {
   const rootElem = document.getElementById("root");
-  rootElem.textContent = `Got ${episodeList.length} episode(s)`;
+  list.forEach((episode) => {
+    let eachDiv = document.createElement(`div`);
+    let containerDiv = rootElem.appendChild(eachDiv);
+    containerDiv.setAttribute(`id`, `${episode.id}`);
+    containerDiv.setAttribute(`class`, `container episode`);
+    containerDiv.setAttribute(`style`, `display: initial;`);
+
+    //take the name of each episode and put it inside containerDiv
+    let nameParagraph = document.createElement(`p`);
+    containerDiv.appendChild(nameParagraph);
+    nameParagraph.setAttribute(
+      `id`,
+      `nameS${codeCorrection(episode.season)}E${codeCorrection(episode.number)}`
```

Comment:
this `codeCorrection` is used many times in a similar way, maybe it would be more readable if the concatenation logic was moved into its own function.

```
function generateSE(episodeOrShow) {
    return `S${codeCorrection(episodeOrShow.season)}E${codeCorrection(episodeOrShow.number)`
}
```

what do you think?